File: LLM-Translations/GPT-4/avatar/Python/Go/atcoder_ABC126_D.go
package main

import "fmt"

type Node struct {
	val    int
	color  bool
	edges  []Edge
}

type Edge struct {
	node *Node
	weight int
}

func traverse(node *Node, distance int, visited map[*Node]bool) {
	_, ok := visited[node]
	if ok {
		return
	} else {
		visited[node] = true
	}
	for _, edge := range node.edges {
		adjNode := edge.node
		dis := edge.weight
		if (distance+dis)%2 == 0 {
			adjNode.color = node.color
		} else {
			adjNode.color = !node.color
		}
		traverse(adjNode, distance+dis, visited)
	}
}

func main() {
	var n int
	fmt.Scan(&n)
	nodes := make([]*Node, n)
	for i := range nodes {
		nodes[i] = &Node{}
	}
	for i := 0; i < n-1; i++ {
		var u, v, w int
		fmt.Scan(&u, &v, &w)
		node1 := nodes[u-1]
		node2 := nodes[v-1]
		node1.val = u
		node2.val = v
		node1.edges = append(node1.edges, Edge{node2, w})
		node2.edges = append(node2.edges, Edge{node1, w})
	}
	root := nodes[0]
	root.color = true
	visited := make(map[*Node]bool)
	traverse(root, 0, visited)
	for i := 0; i < n; i++ {
		node := nodes[i]
		if node.color {
			fmt.Println(1)
		} else {
			fmt.Println(0)
		}
	}
}

End of Code

File: LLM-Translations/GPT-4/avatar/Python/Java/atcoder_ABC126_D.java
import java.util.*;

class Node {
    int val;
    Boolean color;
    List<Pair> a = new ArrayList<>();

    Node(int val) {
        this.val = val;
    }
}

class Pair {
    Node node;
    int weight;

    Pair(Node node, int weight) {
        this.node = node;
        this.weight = weight;
    }
}

public class atcoder_ABC126_D {
    private static Node root;
    private static Set<Node> nodeSet = new HashSet<>();

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        Node[] nodes = new Node[n];
        for (int i = 0; i < n; i++) {
            nodes[i] = new Node(i + 1);
        }
        for (int i = 0; i < n - 1; i++) {
            int u = scanner.nextInt();
            int v = scanner.nextInt();
            int w = scanner.nextInt();
            Node node1 = nodes[u - 1];
            Node node2 = nodes[v - 1];
            node1.a.add(new Pair(node2, w));
            node2.a.add(new Pair(node1, w));
        }
        root = nodes[0];
        root.color = true;
        traverse(root, 0);
        for (Node node : nodes) {
            System.out.println(node.color ? 1 : 0);
        }
    }

    private static void traverse(Node node, int distance) {
        if (nodeSet.contains(node)) {
            return;
        }
        nodeSet.add(node);
        for (Pair pair : node.a) {
            Node adjNode = pair.node;
            int dis = pair.weight;
            if ((distance + dis) % 2 == 0) {
                adjNode.color = root.color;
            } else {
                adjNode.color = !root.color;
            }
            traverse(adjNode, distance + dis);
        }
    }
}

// End of Code.
